core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -132,9 +139,22 @@ class ReplicaFetcherThread(name: String,
 
     brokerTopicStats.updateReplicationBytesIn(records.sizeInBytes)
 
+    log.maybeUpdateHighWatermark(partitionData.highWatermark).foreach { newHighWatermark =>
+      partitionsWithNewHighWatermark += topicPartition
+      if (logTrace)
+        trace(s"Follower received high watermark ${partitionData.highWatermark} from the leader and " +
+          s"updated replica high watermark to $newHighWatermark for partition $topicPartition")
```

Comment:
Should we also log something when the HWM is not changed? Tracing the HWM received from the leader could be useful in some situations. We could perhaps say something like: `Follower received high watermark {} from the leader but did not update replica high watermark for partition {}`.